### PR TITLE
Visa adapter part 3

### PR DESCRIPTION
**Pull request by cpacejo**
Refactoring of internal bind logic to allow for "internal" calls for issuing binds to/from the internal adapter.  (Mostly moved code, not new code.)

Sorry for the big messy review!  Importantly now though -- traffic now passes between the VS adapter and the node's local adapter!

### CODE REVIEW

File: adapter/ph/src/assembly.rs [hunk 1]
```diff
@@ -191,6 +204,56 @@ impl Assembly {
...
+
+        let ingress_tether_id = ingress_peer_state
+            .pft
+            .insert(pep)
+            .map_err(|()| AddRouteError::PftFull)?;
+
+        Ok(ingress_tether_id)
+    }
+
     pub fn hack_default_policy(&self, ingress_link_id: LinkId) -> Option<NonZero<LinkId>> {
```
Comment (mkolehmainen) [inline adapter/ph/src/assembly.rs]: I get that this is a hack of some sort, but would you add a comment about what this is doing?  It's not clear to me how this function relates to a policy.

File: adapter/ph/src/mgmt/handlers.rs [hunk 2]
```diff
@@ -305,7 +304,10 @@ pub async fn handle_bind_agent_address_request(
         dst_port,
     );
 
-    let ingress_link_id = pkt.metadata().ingress_link_id;
+    let Some(ingress_link_id) = NonZero::new(pkt.metadata().ingress_link_id) else {
+        // who sent this??
+        return Ok(());
```
Comment (mkolehmainen) [inline adapter/ph/src/mgmt/handlers.rs:313]: Should this log something? Is this not an error?

File: adapter/ph/src/mgmt/handlers.rs [hunk 3]
```diff
@@ -314,97 +316,73 @@ pub async fn handle_bind_agent_address_request(
...
+                    .await;
+
+                    return Ok(());
+                }
+
+                Err(super::dock::BindAgentAddressError::AddRouteError(
+                    assembly::AddRouteError::PftFull,
+                )) => {
+                    // PFT full; respond with error message
+                    // TODO: maybe tick a counter somewhere?
```
Comment (mkolehmainen) [inline adapter/ph/src/mgmt/handlers.rs:364]: Yes! Let's add project tasks to address these counter needs.

**Review by mkolehmainen — DISMISSED** [review]
Looks good. Made a couple minor commens.

File: adapter/ph/src/mgmt/handlers.rs (hunk 2 above)
Comment (cpacejo) [reply adapter/ph/src/mgmt/handlers.rs:313]: I added a log statement -- for context though this is part of my experiment/campaign to use `Option<NonZero<...>>` types in more places to add some type safety around "unknown" 0 integer values.  (The Rust compiler has special support for this construct such that it compiles the same as a plain integer.)  So this check should go away as I push that pattern into more places.

File: adapter/ph/src/mgmt/handlers.rs (hunk 3 above)
Comment (cpacejo) [reply adapter/ph/src/mgmt/handlers.rs:364]: Added a draft for now will flesh out later!

File: adapter/ph/src/assembly.rs (hunk 1 above)
Comment (cpacejo) [reply adapter/ph/src/assembly.rs]: Done!

**Review by mkolehmainen — APPROVED** [review]
Looks great, thanks!

**Pull request merged**